docs/migration/v12.md

```diff
@@ -0,0 +1,329 @@
+# Platform v12 migration guide: Vite & React 18
```

Comment:
Added in [intro revisions](https://github.com/dhis2/app-platform/pull/884/commits/071f5dc8e22aafd60318ba69ac353da608326cd2), but added v12 since it's helpful to see the relevant package version